<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SceneComposer</class>
 <widget class="QMainWindow" name="SceneComposer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>WorldEditor</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/Images/WorldBuilder.png</normaloff>:/Images/WorldBuilder.png</iconset>
  </property>
  <widget class="QToolWindowManager" name="centralwidget">
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <widget class="HierarchyBrowser" name="hierarchy" native="true">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>10</y>
      <width>200</width>
      <height>200</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>200</width>
      <height>200</height>
     </size>
    </property>
   </widget>
   <widget class="ComponentBrowser" name="components" native="true">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>10</y>
      <width>200</width>
      <height>200</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>200</width>
      <height>200</height>
     </size>
    </property>
   </widget>
   <widget class="ConsoleManager" name="consoleOutput" native="true">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>10</y>
      <width>200</width>
      <height>200</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>200</width>
      <height>200</height>
     </size>
    </property>
   </widget>
   <widget class="ContentBrowser" name="contentBrowser" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>200</width>
      <height>200</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>200</width>
      <height>200</height>
     </size>
    </property>
   </widget>
   <widget class="QWidget" name="projectWidget" native="true">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>220</y>
      <width>200</width>
      <height>200</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="projectLayout">
       <property name="spacing">
        <number>2</number>
       </property>
       <property name="leftMargin">
        <number>2</number>
       </property>
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="rightMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QToolButton" name="commitButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>75</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Commit</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="revertButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>75</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Revert</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="PropertyEditor" name="projectSettings">
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>200</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="propertyWidget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>200</width>
      <height>200</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>200</width>
      <height>200</height>
     </size>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>2</number>
       </property>
       <property name="leftMargin">
        <number>2</number>
       </property>
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="rightMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QToolButton" name="componentButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Add Component</string>
         </property>
         <property name="popupMode">
          <enum>QToolButton::InstantPopup</enum>
         </property>
         <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextBesideIcon</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="PropertyEditor" name="propertyView">
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>200</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="SceneView" name="preview" native="true">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>430</y>
      <width>511</width>
      <height>221</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="viewportWidget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>430</y>
      <width>411</width>
      <height>221</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="viewportLayout">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="orthoButton">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>2D</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="moveButton">
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../res/WorldEditor.qrc">
           <normaloff>:/Images/editor/Move.png</normaloff>:/Images/editor/Move.png</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="autoExclusive">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="rotateButton">
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../res/WorldEditor.qrc">
           <normaloff>:/Images/editor/Rotate.png</normaloff>:/Images/editor/Rotate.png</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="autoExclusive">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="scaleButton">
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../res/WorldEditor.qrc">
           <normaloff>:/Images/editor/Scale.png</normaloff>:/Images/editor/Scale.png</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="autoExclusive">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <widget class="Viewport" name="viewport" native="true">
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>200</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="Timeline" name="timeline" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>660</y>
      <width>411</width>
      <height>111</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_New"/>
    <addaction name="action_Open"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionBuild_Project"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="actionNew_Object"/>
    <addaction name="actionEdit_Object"/>
    <addaction name="actionDelete_Object"/>
    <addaction name="separator"/>
    <addaction name="actionEditor_Mode"/>
    <addaction name="actionGame_Mode"/>
    <addaction name="separator"/>
    <addaction name="actionTake_Screenshot"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
    <addaction name="actionSave_Layout"/>
    <addaction name="actionResore_Layout"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionPlugin_Manager"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionOptions"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menuEdit"/>
   <addaction name="menuTools"/>
   <addaction name="menuWindow"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="action_New">
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_Open">
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="actionContent_Browser">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Content Browser</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="actionObject_properties">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Object properties</string>
   </property>
  </action>
  <action name="actionApplication_Console">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Application Console</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/Images/Undo.png</normaloff>:/Images/Undo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/Images/Redo.png</normaloff>:/Images/Redo.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionEditor_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Editor Mode</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionGame_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Game Mode</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As...</string>
   </property>
  </action>
  <action name="actionScene_objects">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Scene objects</string>
   </property>
  </action>
  <action name="actionEdit_Object">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Edit Object</string>
   </property>
  </action>
  <action name="actionDelete_Object">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Delete Object</string>
   </property>
  </action>
  <action name="actionNew_Object">
   <property name="text">
    <string>New Object</string>
   </property>
  </action>
  <action name="actionComponents">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Components</string>
   </property>
  </action>
  <action name="actionTake_Screenshot">
   <property name="text">
    <string>Take Screenshot</string>
   </property>
   <property name="shortcut">
    <string>F12</string>
   </property>
  </action>
  <action name="actionPlugin_Manager">
   <property name="text">
    <string>About Plugins...</string>
   </property>
   <property name="toolTip">
    <string>Plugin Manager</string>
   </property>
  </action>
  <action name="actionSave_Layout">
   <property name="text">
    <string>Save Layout</string>
   </property>
  </action>
  <action name="actionResore_Layout">
   <property name="text">
    <string>Resore Layout</string>
   </property>
  </action>
  <action name="actionBuild_Project">
   <property name="text">
    <string>Build Project</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="text">
    <string>Options...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PropertyEditor</class>
   <extends>QTreeView</extends>
   <header>editors/propertyedit/propertyeditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QToolWindowManager</class>
   <extends>QWidget</extends>
   <header>managers/toolwindowmanager/qtoolwindowmanager.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ComponentBrowser</class>
   <extends>QWidget</extends>
   <header>editors/componentbrowser/componentbrowser.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ContentBrowser</class>
   <extends>QWidget</extends>
   <header>editors/contentbrowser/contentbrowser.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>HierarchyBrowser</class>
   <extends>QWidget</extends>
   <header>editors/objecthierarchy/hierarchybrowser.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ConsoleManager</class>
   <extends>QWidget</extends>
   <header>managers/consolemanager/consolemanager.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SceneView</class>
   <extends>QWidget</extends>
   <header>graph/sceneview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Viewport</class>
   <extends>QWidget</extends>
   <header>graph/viewport.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Timeline</class>
   <extends>QWidget</extends>
   <header>editors/timeline/timeline.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../res/WorldEditor.qrc"/>
 </resources>
 <connections/>
</ui>
